### PR TITLE
Altcoins: remove obsolete override, update some altcoins

### PR DESCRIPTION
**Pull request by Lassulus**
###### Motivation for this change

remove the obsolete override in all-packages.nix.
fix broken altcoins.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Infinisil** [comment]
Regarding namecoin: I was just about to open a PR for the update: https://github.com/Infinisil/nixpkgs/commit/bb6fa63d85aa18cbd5cb15a99b8c7baf4f8e6e41

I'm gonna comment on the differences

File: pkgs/applications/altcoins/namecoin.nix [hunk 1]
```diff
@@ -1,20 +1,32 @@
...
+  buildInputs = [ openssl db4 boost libtool libuuid
+                  protobuf libevent ]
+                  ++ optionals withGui [ qt4 ];
 
-  installPhase = ''
-    mkdir -p $out/bin
-    cp namecoin-qt $out/bin
-  '';
+  configureFlags = [ "--with-boost-libdir=${boost.out}/lib" ]
+                     ++ optionals withGui [ "--with-gui=qt4" ];
```
Comment (Infinisil) [inline pkgs/applications/altcoins/namecoin.nix]: `--with-gui=qt4` is not needed, it gets detected automatically

File: pkgs/applications/altcoins/namecoin.nix [hunk 2]
```diff
@@ -1,20 +1,32 @@
...
+    repo = "namecoin-core";
+    rev = version;
+    sha256 = "17zz0rm3js285w2assxp8blfx830rs0ambcsaqqfli9mnaik3m39";
+  };
 
-  qmakeFlags = [ "USE_UPNP=-" ];
+  nativeBuildInputs = [ pkgconfig autoreconfHook ];
+  buildInputs = [ openssl db4 boost libtool libuuid
+                  protobuf libevent ]
+                  ++ optionals withGui [ qt4 ];
```
Comment (Infinisil) [inline pkgs/applications/altcoins/namecoin.nix]: db4 and protobuf are [only needed for the GUI](https://github.com/namecoin/namecoin-core/blob/6e7eb2678e5a12a84a086e8079807094e0db9a74/doc/build-unix.md#dependencies)

There are also some other different dependencies between our versions.

File: pkgs/applications/altcoins/namecoin.nix [hunk 3]
```diff
@@ -1,20 +1,32 @@
...
-    mkdir -p $out/bin
-    cp namecoin-qt $out/bin
-  '';
+  configureFlags = [ "--with-boost-libdir=${boost.out}/lib" ]
+                     ++ optionals withGui [ "--with-gui=qt4" ];
 
-  meta = namecoind.meta;
+  meta = {
+    description = "Decentralized open source information registration and transfer system based on the Bitcoin cryptocurrency";
+    homepage = http://namecoin.info;
```
Comment (Infinisil) [inline pkgs/applications/altcoins/namecoin.nix]: Now https://namecoin.org

File: pkgs/applications/altcoins/namecoin.nix (hunk 2 above)
Comment (Infinisil) [reply pkgs/applications/altcoins/namecoin.nix]: Sorry, db4 is needed in any case, but there is the optional libqrencode dependency

**Comment by Lassulus** [comment]
@Infinisil thanks for the comments, I'm gonna test them and update the PR accordingly.
Sorry for making your work obsolete, but I had a PR ready since yesterday

**Comment by Lassulus** [comment]
@Infinisil I cherry-picked your commit. If you make any new updates just give me a message :)

**Comment by Infinisil** [comment]
Thanks :)

**Pull request merged**